mlir/lib/Dialect/MIOpen/utility/loweringUtils.cpp

```diff
@@ -27,6 +27,7 @@ void propagateTransformOob(TransformMapAttr transformMap,
 
     switch (transform.getType()) {
     case TransformType::PassThrough:
+    case TransformType::Broadcast: // is this right??
```

Comment:
No, no it isn't.

The index diff update rule fro broadcast is as follows: `lowerDiff[lowerDims[i]] = upperDiff[upperDims[i]] % params[i]`

That is, suppose you've broadcast `[a, b]` into `[a, b, a, b, ...]` Then, adding 1 moves you to the other element, but adding 2 keeps where you are.